check for missing migrations in build

snopoke opened the pull request.

snopoke committed:
check for missing migrations in build

gherceg reviewed (commented):
Nice addition. I probably should know this, but I'm not sure what kicks this off in our build process and would appreciate any extra info on that.

snopoke commented:
@gherceg our build process is a bit convoluted. When you run the `./scripts/docker test` command this results in the following command being run from inside that script:
```
docker-compose run --rm web run_tests "${TEST:-python}" "$@"
```

Looking at the [dockerfile](https://github.com/dimagi/commcare-hq/blob/c17de39923da6f6cfe0b99e158f49bb53c6b379f/docker/Dockerfile#L8) for the `web` container we see that it uses `run.sh` as the entrypoint which will do some setup and then simply execute whatever arguments are passed to it (["$@"](https://github.com/dimagi/commcare-hq/blob/4d241158b498d621a8cb3f7f10f579cc9fdec23d/docker/run.sh#L188)).

The `run_tests` command is a bash function defined in the `run.sh` file with does further setup and then calls the actual `run_test.sh` script (which, [through some bash magic](https://github.com/dimagi/commcare-hq/blob/4d241158b498d621a8cb3f7f10f579cc9fdec23d/docker/run.sh#L155),  is created as part of the `run.sh` setup.

snopoke committed:
Merge branch 'master' into sk/test-migrations

orangejenny committed:
Merge branch 'master' into sk/test-migrations

snopoke committed:
set DB names for citus in docker

This is needed so that Django can find the DB since we only create the
test DB during docker tests and not the non-test DB.

snopoke committed:
Merge branch 'sk/test-migrations' of github.com:dimagi/commcare-hq into sk/test-migrations

snopoke commented:
now waiting on #28565 and https://github.com/dimagi/commcare-icds/pull/118

snopoke committed:
Merge branch 'master' into sk/test-migrations
